src/nbl/video/COpenGLCommandPool.cpp

```diff
@@ -0,0 +1,181 @@
+{
+    if (m_maxDrawCount == 0u)
+        return;
+
+    ctxlocal->nextState.vertexInputParams.indirectDrawBuf = core::smart_refctd_ptr_static_cast<const COpenGLBuffer>(m_buffer);
+    ctxlocal->nextState.vertexInputParams.parameterBuf = core::smart_refctd_ptr_static_cast<const COpenGLBuffer>(m_countBuffer);
+
+    ctxlocal->flushStateGraphics(gl, SOpenGLContextLocalCache::GSB_ALL, ctxid);
+
+    GLenum idxType = GL_INVALID_ENUM;
+    switch (ctxlocal->currentState.vertexInputParams.vaoval.idxType)
+    {
+    case asset::EIT_16BIT:
+        idxType = GL_UNSIGNED_SHORT;
+        break;
+    case asset::EIT_32BIT:
+        idxType = GL_UNSIGNED_INT;
+        break;
+    default:
+        break;
+    }
+
+    const asset::E_PRIMITIVE_TOPOLOGY primType = ctxlocal->currentState.pipeline.graphics.pipeline->getRenderpassIndependentPipeline()->getPrimitiveAssemblyParams().primitiveType;
+    GLenum glpt = COpenGLCommandBuffer::getGLprimitiveType(primType);
+
+    GLuint64 offset = m_offset;
+    static_assert(sizeof(offset) == sizeof(void*), "Bad reinterpret_cast");
+    gl->extGlMultiDrawElementsIndirectCount(glpt, idxType, reinterpret_cast<void*>(offset), m_countBufferOffset, m_maxDrawCount, m_stride);
+}
+
+void COpenGLCommandPool::CBeginRenderPassCmd::operator()(IOpenGL_FunctionTable* gl, SOpenGLContextLocalCache* ctxlocal, uint32_t ctxid, const system::logger_opt_ptr logger, const COpenGLCommandBuffer* executingCmdbuf)
+{
+    auto framebuf = core::smart_refctd_ptr_static_cast<const video::COpenGLFramebuffer>(m_framebuffer);
+
+    ctxlocal->nextState.framebuffer.hash = framebuf->getHashValue();
+    ctxlocal->nextState.framebuffer.fbo = std::move(framebuf);
+    ctxlocal->flushStateGraphics(gl, SOpenGLContextLocalCache::GSB_FRAMEBUFFER, ctxid);
+
+    GLuint fbo = ctxlocal->currentState.framebuffer.GLname;
+    if (fbo)
+    {
+        IGPUCommandBuffer::SRenderpassBeginInfo beginInfo = {};
+        beginInfo.clearValueCount = m_clearValueCount;
+        beginInfo.clearValues = m_clearValues;
+        beginInfo.framebuffer = m_framebuffer;
+        beginInfo.renderArea = m_renderArea;
+        beginInfo.renderpass = m_renderpass;
+        COpenGLCommandBuffer::beginRenderpass_clearAttachments(gl, ctxlocal, ctxid, beginInfo, fbo, logger);
+    }
+
+    executingCmdbuf->currentlyRecordingRenderPass = m_renderpass.get();
+}
+
+void COpenGLCommandPool::CEndRenderPassCmd::operator()(IOpenGL_FunctionTable* gl, SOpenGLContextLocalCache* ctxLocal, uint32_t ctxid, const system::logger_opt_ptr logger, const COpenGLCommandBuffer* executingCmdbuf)
+{
+    ctxLocal->nextState.framebuffer.hash = SOpenGLState::NULL_FBO_HASH;
+    ctxLocal->nextState.framebuffer.GLname = 0u;
+    ctxLocal->nextState.framebuffer.fbo = nullptr;
+    executingCmdbuf->currentlyRecordingRenderPass = nullptr;
+}
```

Comment:
not gonna review since its subject to a lot of potential change

Reply:
Resolved.